### PR TITLE
RSE-50: fix: allowing user to see logs after the job is deleted

### PR DESCRIPTION
**Pull request by Oalbarran94**
**##### Problem**

When you delete a job from the job list page, and you try to see the log output of that job executions, an empty log is shown (it only happens when deleting the job from that page)

**##### Solution**

It was added a way to know if the job was deleted, if it was, Rundeck is going to use the path stored for that job in the execution table.

fix [RSE-50](https://pagerduty.atlassian.net/browse/RSE-50)
fix https://github.com/rundeckpro/rundeckpro/issues/2354

With this fix the user can watch the logs for a specific job even when this was deleted. The previous behavior did not allow to watch those logs even though they were stored in the file system. 

Execution panel with job deleted: 
deletedJob

Activity screen for deleted job: 
activityScreen

Logs from deleted job:
logDeleted

This fix was also tested with a custom file system as minio 

Screen Shot 2022-09-26 at 16 47 20

### CODE REVIEW

**Comment by Oalbarran94** [comment]
@Mergifyio rebase

**Comment by Oalbarran94** [comment]
retest this please

File: rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy [hunk 1]
```diff
@@ -1184,7 +1184,7 @@ class LogFileStorageService
             boolean checkPartial = false
     )
     {
-        File file = getFileForExecutionFiletype(execution, filetype, false, false)
+        File file = getFileForExecutionFiletype(execution, filetype, execution.scheduledExecution == null, false)
```
Comment (L2JE) [inline rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy:1187]: Use a variable for all these `execution.scheduledExecution == null` in different calls, (use a good name that shows why you need it)

File: rundeckapp/src/test/groovy/rundeck/services/LogFileStorageServiceSpec.groovy [hunk 2]
```diff
@@ -1016,11 +1031,10 @@ class LogFileStorageServiceSpec extends RundeckHibernateSpec implements ServiceU
 
             def exec = new Execution(
                     dateStarted: new Date(),
-                    dateCompleted: null,
+                    dateCompleted: new Date(),
                     user: 'user2',
                     project: 'test',
-                    serverNodeUUID: 'D0CA0A6D-3F85-4F53-A714-313EB57A4D1F',
-                    outputfilepath: '/tmp/file'
+                    serverNodeUUID: 'D0CA0A6D-3F85-4F53-A714-313EB57A4D1F'
```
Comment (L2JE) [inline rundeckapp/src/test/groovy/rundeck/services/LogFileStorageServiceSpec.groovy:1037]: Why did you remove `exec.outputFilePath` in this and the following tests _(L1100, L1164)_?

**Review by L2JE — CHANGES_REQUESTED** [review]
If possible, add the issue scenario _"Execution log not found after job deletion"_ as a test case. (If not add a comment explaining why)

File: rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy (hunk 1 above)
Comment (Oalbarran94) [reply rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy:1187]: We are not supposed to get "Execution log not found after job deletion" since the goal of the card was to show the logs even after a job was deleted. This is possible because Rundeck is saving all executions pointing to where the logs are located.

File: rundeckapp/src/test/groovy/rundeck/services/LogFileStorageServiceSpec.groovy (hunk 2 above)
Comment (Oalbarran94) [reply rundeckapp/src/test/groovy/rundeck/services/LogFileStorageServiceSpec.groovy:1037]: As we changed the way of validating (execution.scheduleExecution == null) if the execution was going to get the file by getting it from "outputfilepath" or by forming the filepath (as showed below) , that part of the test needed to be remove in order to follow its normal steps. Since within those test we do not use a valid execution.scheduleExecution, the tests itself were getting another path which they were not created for.

image

File: rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy (hunk 1 above)
Comment (L2JE) [reply rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy:1187]: What I mean by covering that use case is to make a test that verifies if the logs are still accessible after deletion trying to use the interfaces that the application is currently using to show them on the screen.

Comment (alexander-variacode) [reply rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy:1187]: Osman try to save this operation `execution.scheduledExecution == null` in a variable because It just like that it didn’t explain why is there.

Comment (ronaveva) [reply rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy:1187]: this scenario would still be possible if someone just deletes the log file manually, but i think that is covered by the tests 

https://github.com/rundeck/rundeck/blob/fff68b5a30fd3d06b611e011e8e19ece38b14c1a/rundeckapp/src/test/groovy/rundeck/services/LogFileStorageServiceSpec.groovy#L802

Comment (L2JE) [reply rundeckapp/grails-app/services/rundeck/services/LogFileStorageService.groovy:1187]: Please confirm that with the current tests, in case the logs of deleted jobs don't appear again as before, the tests will complain about it.

**Review by L2JE — APPROVED** [review]
LGTM

**Review by alexander-variacode — APPROVED** [review]
LGTM

**Review by ronaveva — APPROVED** [review]
LGTM
